User table, migrations, seeds, userId endpoint created.

CookieMonsta89 opened the pull request.

tgreenhalgh commented:
Please use the comment template for every PR:

# Description

Enter the description here.

## Type of change

---

What did you change?

# How Has This Been Tested?

How was this tested?

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

CookieMonsta89 commented:
Created migrations for users table.  Used faker.js to generate phony data for users table. We also created an endpoint to retrieve data for a specific user.

CookieMonsta89 commented:
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
faker.js
Fixes # (issue)

## Type of change
Created a table for users using migrations.  We also added seeds along with faker.js to create phony data for users.  We then created an endpoint that allowed for us to retrieve users data by id. 
Please delete options that are not relevant.

- [x ] New feature (non-breaking change which adds functionality)
- [x ] This change requires a documentation update

## Change status
- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?
Using postman and insomnia
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
-we used dbbrowser in order to test that faker.js added user data to table.  To test endpoint, the server must be up and running on localhost:9001.  The url is localhost:9001/api/user/info/1.  We tested endpoint by using postman as well as insomnia. 
- [x ] Test A
- [x ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] My code has been reviewed by at least one peer
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [x ] I have added tests that prove my fix is effective or that my feature works
